### ISSUE TITLE
RemoveOrder doesn't find the order

### ISSUE THREAD

**Issue by moshababo** [ExchangeUnion/xud#804]
From `xud-simulation`:

```
08/02/2019 09:55:54.806 [RPC] debug: received call /xudrpc.Xud/PlaceOrderSync
08/02/2019 09:55:54.808 [ORDERBOOK] debug: order added: {"pairId":"LTC/BTC","price":10,"quantity":10,"isBuy":true,"localId":"random_order_id","hold":0,"id":"f6314470-2b76-11e9-9d4b-2144188b75ef","initialQuantity":10,"createdAt":1549612554807}
08/02/2019 09:55:54.814 [RPC] debug: received call /xudrpc.Xud/GetInfo
08/02/2019 09:55:54.821 [RPC] debug: received call /xudrpc.Xud/GetInfo
08/02/2019 09:55:56.225 [RPC] debug: received call /xudrpc.Xud/RemoveOrder
08/02/2019 09:55:56.227 [ORDERBOOK] error: attempted to remove non-existing orderId (f6314470-2b76-11e9-9d4b-2144188b75ef)
```
The issue was introduced by this commit: https://github.com/ExchangeUnion/xud/commit/e46d55d8528ba311a5fb8719dfe1c32e574efa03

In addition, the response to `RemoveOrder` doesn't return an error, and so the error was silent. This problem existed before - `removeOwnOrderByLocalId` doesn't throw an error when `removeOwnOrder` fails and returns false. The later can be changed to throw an error instead of returning a boolean.

**Comment by kilrau** [comment]
Since it was your commit - are you taking care? @sangaman

**Comment by sangaman** [comment]
This is fixed by #809. I'll see about preventing any silent errors in a separate PR, for now though if the order actually doesn't exist you'll get a `Error: 5 NOT_FOUND: order with local id 33da05f0-3482-11e9-97b6-e35671d0c612 does not exist`.
